### PR TITLE
Feature/ctech 2300 run notebook runner tests on only notebooks that have been added or modified in sample notebooks

### PR DESCRIPTION
**Pull request by adamMQM**
# Pull Request Checklist

- [x] Changes follow the [style guide](https://github.com/finbourne/sample-notebooks/blob/master/docs/FINBOURNE%20notebook%20style%20guide.ipynb)
- [x] Tests pass
- [x] Raised the PR against the `develop` branch

# Description of the PR

Since we've added the option to test notebooks by passing filepaths to the test runner, we can now run tests on only notebooks that have been added or modified on pull-request

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -24,7 +25,10 @@ services:
...
         export FBN_NOTEBOOK_FOLDER=/usr/notebooks
+        # get all files that have been:
+        # Added (A), Copied (C), Modified (M), Renamed (R), have their type (i.e. regular file, symlink, submodule, …​) changed (T)
+        # between tip of merge and develop branch
+        export GIT_DIFF=$(git diff --name-only --diff-filter=ACMRT $DEVELOP_SHA HEAD | xargs)
         cd $${FBN_NOTEBOOK_FOLDER}
         pip install -r requirements.txt
-        python /usr/src/nbrunner.py --notebooks $${FBN_NOTEBOOK_FOLDER}/examples -c
-        python /usr/src/nbrunner.py --notebooks $${FBN_NOTEBOOK_FOLDER}/university -c
+        python /usr/src/nbrunner/main.py -f $GIT_DIFF -d ' ' -c
```
[the remark this answers is not in the document]
Comment (adamMQM) [reply docker-compose.yml:34]: multiple, `git diff --name-only --diff-filter=ACMRT $DEVELOP_SHA HEAD | xargs` produces a space separated list of notebooks, and `-d`  specifies the delimiter

**Review by rizwansaeed — DISMISSED** [review]

**Comment by rizwansaeed** [comment]
build failed https://github.com/finbourne/sample-notebooks/actions/runs/3901496784/jobs/6663412790

**Comment by adamMQM** [comment]
> build failed https://github.com/finbourne/sample-notebooks/actions/runs/3901496784/jobs/6663412790

Will work when new version of nbrunner is deployed

**Comment by adamMQM** [comment]
tested locally with candidate version and test sha's:

Screenshot 2023-01-12 at 11 54 29

**Review by rizwansaeed — APPROVED** [review]

File: docker-compose.yml [hunk 2]
```diff
@@ -25,6 +24,11 @@ services:
         pip config set global.disable-pip-version-check true
         export FBN_NOTEBOOK_FOLDER=/usr/notebooks
         cd $${FBN_NOTEBOOK_FOLDER}
+        # get all files that have been:
+        # Added (A), Copied (C), Modified (M), Renamed (R), have their type (i.e. regular file, symlink, submodule, …​) changed (T)
+        # between tip of merge and develop branch
+        # export GIT_DIFF=$$(git diff --name-only --diff-filter=ACMRT origin/develop..HEAD)
```
Comment (rizwansaeed) [inline docker-compose.yml:30]: can we delete the commented code?

File: docker-compose.yml [hunk 3]
```diff
@@ -25,6 +24,11 @@ services:
...
         pip install -r requirements.txt
-        python /usr/src/nbrunner.py --notebooks $${FBN_NOTEBOOK_FOLDER}/examples -c
-        python /usr/src/nbrunner.py --notebooks $${FBN_NOTEBOOK_FOLDER}/university -c
+        # python /usr/src/nbrunner/main.py -f "$$GIT_DIFF" -d $$'\n' -c
```
Comment (rizwansaeed) [inline docker-compose.yml:32]: and this one?

File: docker-compose.yml (hunk 2 above)
Comment (adamMQM) [reply docker-compose.yml:30]: We can do, although I will follow up this PR shortly with another to run only the notebooks that have change. The next PR will just uncomment that code and delete the rest.

File: docker-compose.yml (hunk 3 above)
Comment (adamMQM) [reply docker-compose.yml:32]: same as above

**Review by rizwansaeed — APPROVED** [review]

**Pull request merged**